AsyncDisplayKit/ASVideoNode.mm

```diff
@@ -0,0 +1,352 @@
+
+
+#import "ASVideoNode.h"
+
+@interface ASVideoNode ()
+{
+  ASDN::RecursiveMutex _lock;
+  
+  __weak id<ASVideoNodeDelegate> _delegate;
+
+  BOOL _shouldBePlaying;
+  AVAsset *_asset;
+  
+  AVPlayerItem *_currentItem;
+  AVPlayer *_player;
+  
+  ASButtonNode *_playButton;
+  ASDisplayNode *_playerNode;
+  ASDisplayNode *_spinner;
+  ASVideoGravity _gravity;
+}
+
+@end
+
+@interface ASDisplayNode ()
+- (void)setInterfaceState:(ASInterfaceState)newState;
+@end
+
+@implementation ASVideoNode
+
+- (instancetype)init
+{
+  if (!(self = [super init])) { return nil; }
+  
+  self.gravity = ASVideoGravityResizeAspect;
+  
+  [self addTarget:self action:@selector(tapped) forControlEvents:ASControlNodeEventTouchUpInside];
+  
+  [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(didPlayToEnd:) name:AVPlayerItemDidPlayToEndTimeNotification object:nil];
```

Comment:
Can you move this notification subscription to a later point when it is more likely to be required, even if that is -didLoad?  We may allocate thousands of nodes at once for measurement for table heights.

Also make sure we unsubscribe in -dealloc, but unsubscribe from the specific notification rather than all, as there is a pretty big efficiency boost from not having to scan all registered types of notifications for self as a possible observer.